poetry/core/masonry/utils/include.py

```diff
@@ -46,3 +46,24 @@ def refresh(self):  # type: () -> Include
         self._elements = sorted(list(self._base.glob(self._include)))
 
         return self
+
+
+class IncludeFile:
+    def __init__(
+        self,
+        path,  # type: Path
+        source_root=None,  # type: Optional[Path]
+    ):
+        self.path = path
+        self.source_root = source_root or Path(".")
+
+    def __repr__(self):  # type: () -> str
+        return str(self.path)
+
+    @property
+    def rel_path(self):  # type: () -> Path
+        return self.path
+
+    @property
+    def full_path(self):  # type: () -> Path
+        return self.source_root / self.path
```

Comment:
```suggestion
    def relative_path(self):  # type: () -> Path
        if self.parent:
            return self.path.resolve().relative_to(self.parent.resovle())
        return self.path

    def resolve(self):  # type: () -> Path
        if self.parent:
            return (self.parent / self.path).resolve()
        return self.path.resolve()
```

Something like this might be a cleaner approach.

Reply:
I'm not sure I understand the example for `relative_path`.

`self.path`, which is usually the relative path, when resolved relative to its parent because it resolves itself to whatever the current directory is (in the example below, it is my `poetry-core` directory: `/Users/stephsamson/Code/opensource/poetry-core/`) instead of the actual parent directory (`'/Users/stephsamson/Code/opensource/poetry-core/tests/masonry/builders/fixtures/with_include_inline_table'`).

example:
```
ValueError:  '/Users/stephsamson/Code/opensource/poetry-core/LICENSE' does not start with '/Users/stephsamson/Code/opensource/poetry-core/tests/masonry/builders/fixtures/with-include'
```